packages/x-components/src/x-modules/search/components/partial-query-button.vue

```diff
@@ -22,41 +23,54 @@
    *
    * @public
    */
-  @Component({
-    mixins: [xComponentMixin(searchXModule)]
-  })
-  export default class PartialQueryButton extends Vue {
-    /**
-     * The query property.
-     *
-     * @public
-     */
-    @Prop({ required: true })
-    public query!: string;
-
-    /**
-     * Generates the {@link WireMetadata} object omitting the moduleName.
-     *
-     * @returns The {@link WireMetadata} object omitting the moduleName.
-     * @internal
-     */
-    protected createEventMetadata(): Omit<WireMetadata, 'moduleName'> {
-      return {
-        target: this.$el as HTMLElement,
+  export default defineComponent({
+    name: 'PartialQueryButton',
+    xModule: searchXModule.name,
+    props: {
+      /**
+       * The query property.
+       *
+       * @public
+       */
+      query: {
+        type: String,
+        required: true
+      }
+    },
+    setup(props) {
+      useRegisterXModule(searchXModule);
+
+      const $x = use$x();
+
+      const partialButtonEl = ref<HTMLElement>();
+
+      /**
+       * Generates the {@link WireMetadata} object omitting the moduleName.
+       *
+       * @returns The {@link WireMetadata} object omitting the moduleName.
+       * @internal
+       */
+      const createEventMetadata = (): Omit<WireMetadata, 'moduleName'> => ({
+        target: partialButtonEl.value as HTMLElement,
         feature: 'partial_result'
+      });
+
+      /**
+       * Emits events when the button is clicked.
+       *
+       * @public
+       */
+      const emitEvents = () => {
+        $x.emit('UserAcceptedAQuery', props.query, createEventMetadata());
+        $x.emit('UserClickedPartialQuery', props.query, createEventMetadata());
       };
-    }
 
-    /**
-     * Emits events when the button is clicked.
-     *
-     * @public
-     */
-    protected emitEvents(): void {
-      this.$x.emit('UserAcceptedAQuery', this.query, this.createEventMetadata());
-      this.$x.emit('UserClickedPartialQuery', this.query, this.createEventMetadata());
+      return {
+        emitEvents,
+        partialButtonEl
```

Comment:
```suggestion
        partialButtonEl,
        emitEvents
```

Following pattern vars + methods